### PR TITLE
docs: Migrate link formats

### PR DESCRIPTION
**Pull request by ashleemboyer**
## What

- Migrates the formats of links in Markdown content as part of the [Docs Content Link Rewrites project](https://docs.google.com/document/d/1WaSyvoVPS-YCNiSPX0ynGpvc1gySEcpbrRoQVhimQCA/edit)
- Adds two new GitHub workflow files:
  - `.github/workflows/check-legacy-links-format.yml` - This is a workflow that checks if links in markdown content are in the new format. It does this by running the link rewrite script from `dev-portal`, and then asserting that no "rewriteable" links were found. This workflow will be used on subsequent PRs to maintain the new link format.
  - `.github/workflows/test-link-rewrites.yml` - This is a workflow that executes e2e tests of this PR's changes against the `main` branch's deploy preview. The tests only run for branches named `docs/amb.migrate-link-formats`, so future PRs will not have the tests run against them. This is a temporary workflow.

## PRs that came before this as pre-work

- https://github.com/hashicorp/packer/pull/12196

## Reviewing

With the two new workflows in place, this PR should only need a spot-check.

## Testing

- [ ] The latest run of the "Legacy Link Format Checker" workflow for this branch should pass ([check here](https://github.com/hashicorp/packer/actions/workflows/check-legacy-links-format.yml))
- [ ] The latest run of the "Test Link Rewrites" workflow for this branch should pass ([check here](https://github.com/hashicorp/packer/actions/workflows/test-link-rewrites.yml))

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1203453044319637

### CODE REVIEW

File: website/content/partials/post-processor/manifest/Config-not-required.mdx
Comment (kendallstrautman) [inline website/content/partials/post-processor/manifest/Config-not-required.mdx:12]: 🤔 Hmm I wonder if we should just not edit this since it seems to be connected with the product output. @ashleemboyer I think that's what's making[ this test fail. ](https://github.com/hashicorp/packer/actions/runs/3972181961/jobs/6809819588)

Comment (kendallstrautman) [reply website/content/partials/post-processor/manifest/Config-not-required.mdx:12]: Especially since the comment above says 'DO NOT EDIT MANUALLY -->'  😆

Comment (kendallstrautman) [reply website/content/partials/post-processor/manifest/Config-not-required.mdx:12]: Or maybe we update it in the product... I see it here https://github.com/hashicorp/packer/blob/main/post-processor/manifest/post-processor.go#L35

Comment (kendallstrautman) [reply website/content/partials/post-processor/manifest/Config-not-required.mdx:12]: I'll ask the packer team

Comment (ashleemboyer) [reply website/content/partials/post-processor/manifest/Config-not-required.mdx:12]: That's what I was going to suggest! 👍🏻

Comment (kendallstrautman) [reply website/content/partials/post-processor/manifest/Config-not-required.mdx:12]: So I got confirmation to update it in the source. My thought was to use this absolute path, `https://developer.hashicorp.com/packer/docs/templates/legacy_json_templates/engine` but I'm worried the rewrite script will want to make this relative and the tests will fail. Trying to think of the best path forward

Comment (kendallstrautman) [reply website/content/partials/post-processor/manifest/Config-not-required.mdx:12]: Perhaps the relative path is okay though, since this comment is meant to be used to generate the docs.

Comment (ashleemboyer) [reply website/content/partials/post-processor/manifest/Config-not-required.mdx:12]: I think a relative path will be ok too!

File: website/content/docs/templates/hcl_templates/blocks/build/provisioner.mdx
Comment (kendallstrautman) [inline website/content/docs/templates/hcl_templates/blocks/build/provisioner.mdx:138]:
```suggestion
HCL attribute map where the key is the name of a [source
```
To resolve with updates from `main`

Comment (kendallstrautman) [reply website/content/docs/templates/hcl_templates/blocks/build/provisioner.mdx:138]: to not overwrite https://github.com/hashicorp/packer/pull/12220, must have been messed up in the merge

Comment (kendallstrautman) [reply website/content/docs/templates/hcl_templates/blocks/build/provisioner.mdx:138]: To not overwrite https://github.com/hashicorp/packer/commit/ba546eb8cbeed152a4c1211b7022f32e78cdefe3, must been messed up in the merge

Comment (nywilken) [reply website/content/docs/templates/hcl_templates/blocks/build/provisioner.mdx]: Good catch.

**Comment by nywilken** [comment]
Failing test appear to be the results of a bug introduced in the SDK https://github.com/rossmaclean/packer-plugin-sdk/blob/ab2cb594fed2a61571cdd4971dbc410fa8b2ef64/shell-local/config.go#L153; the change was introduced in https://github.com/hashicorp/packer-plugin-sdk/pull/149

**Comment by nywilken** [comment]
> Failing test appear to be the results of a bug introduced in the SDK https://github.com/rossmaclean/packer-plugin-sdk/blob/ab2cb594fed2a61571cdd4971dbc410fa8b2ef64/shell-local/config.go#L153; the change was introduced in [hashicorp/packer-plugin-sdk#149](https://github.com/hashicorp/packer-plugin-sdk/pull/149)

This has been fixed in the v0.3.4 of the SDK, which has been committed to this branch.

**Review by nywilken — APPROVED** [review]
@kendallstrautman I believe this is all good to go now. I'm going to approve in its current state to unblock and allow you to merge when ready. If we need to address any other generated docs or SDK changes please feel free to ping me directly.

File: website/content/community-tools.mdx
Comment (ashleemboyer) [inline website/content/community-tools.mdx:22]: To update after https://github.com/hashicorp/dev-portal/pull/1578 is merged

```suggestion
check out the docs on [extending Packer](https://developer.hashicorp.com/packer/docs/plugins/install-plugins)
```

Comment (kendallstrautman) [reply website/content/community-tools.mdx:22]: As discussed in slack, we will fix in a follow up PR!

**Comment by lbajolet-hashicorp** [comment]
Looks like the new actions are failing on `main` after the extra link migration @kendallstrautman opened a PR for earlier today (https://github.com/hashicorp/packer/pull/12225), any idea what we can do to make the CI green again?
I assume tests didn't catch it before merging since it wasn't based on top of `main` initially.

**Comment by ashleemboyer** [comment]
@lbajolet-hashicorp I'm looking into this now! Thank you for the tag 💖

**Comment by ashleemboyer** [comment]
@lbajolet-hashicorp hi again! This is an update to let you know the fix for this has been merged in [dev-portal#1583](https://github.com/hashicorp/dev-portal/pull/1583), and I've verified that it works on the `main` branch here. 🥳 

https://github.com/hashicorp/packer/actions/workflows/check-legacy-links-format.yml?query=branch%3Amain

**Comment by lbajolet-hashicorp** [comment]
Great news, thanks @ashleemboyer !

**Pull request merged**
